### PR TITLE
[CMSIS-NN] Update microNPU demo to include offloading to CMSIS-NN

### PR DESCRIPTION
**Pull request by grant-arm**
This PR updates the microNPU demo to include offloading to CMSIS-NN.
In particular, the demo is updated to use Mobilenet v2 and the softmax function is offloaded to CMSIS-NN, while the Conv2D's and DepthwiseConv2D's are offloaded to the microNPU.

@Mousius @manupa-arm @areusch @ashutosh-arm

### CODE REVIEW

File: apps/microtvm/ethosu/README.md [hunk 1]
```diff
@@ -72,15 +73,15 @@ the locations for these can be specified as arguments to run_demo.sh, for exampl
 ```
 
 This will:
-- Download a quantized mobilenet v1 model
-- Use tvmc to compile the model for Cortex(R)-M55 CPU and Ethos(TM)-U55 NPU
-- Download an image of a kitten to run the model on
+- Download a quantized (int8) mobilenet v2 model
+- Use tvmc to compile the model for Cortex(R)-M55 CPU, Ethos(TM)-U55 NPU and CMSIS-NN
+- Download an image of a penguig to run the model on
```
Comment (Mousius) [inline apps/microtvm/ethosu/README.md:78]: I don't think this hybrid is the one we're testing for.

File: apps/microtvm/ethosu/Makefile [hunk 2]
```diff
@@ -57,6 +59,7 @@ CODEGEN_SRCS = $(wildcard $(abspath $(BUILD_DIR))/codegen/host/src/*.c)
 CODEGEN_OBJS = $(subst .c,.o,$(CODEGEN_SRCS))
 CMSIS_STARTUP_SRCS = $(wildcard ${CMSIS_PATH}/Device/ARM/${ARM_CPU}/Source/*.c)
 UART_SRCS = $(wildcard ${CORSTONE_300_PATH}/*.c)
+CMSIS_NN_LIBS = $(wildcard ${CMSIS_PATH}/CMSIS/NN/build/Source/*/*.a)
```
Comment (Mousius) [inline apps/microtvm/ethosu/Makefile:62]: For the purposes of the demo we should reference the `.c` files and not rely on these being built externally.

File: apps/microtvm/ethosu/run_demo.sh [hunk 3]
```diff
@@ -123,32 +123,32 @@ script_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
 mkdir -p build
 cd build
 
-# Get mobilenet_v1 tflite model
-mobilenet_url='https://storage.googleapis.com/download.tensorflow.org/models/mobilenet_v1_2018_08_02/mobilenet_v1_1.0_224_quant.tgz'
-curl --retry 64 -sSL ${mobilenet_url} | gunzip | tar -xvf - ./mobilenet_v1_1.0_224_quant.tflite
+# Get mobilenet_v2 tflite model
+mobilenet_url='https://github.com/ARM-software/ML-zoo/raw/master/models/image_classification/mobilenet_v2_1.0_224/tflite_int8/mobilenet_v2_1.0_224_INT8.tflite'
```
Comment (Mousius) [inline apps/microtvm/ethosu/run_demo.sh:127]: We recently had issues with referencing `master` (https://github.com/apache/tvm/pull/9887) and the model changing, can you pick a SHA1 that guarantees we'll always get the same model?

**Review by Mousius — CHANGES_REQUESTED** [review]
This is such an exciting improvement @grant-arm! Really interesting to see a network running across several different pieces of TVM at once :smile_cat: It looks pretty good to me, just a few comments.

File: apps/microtvm/ethosu/README.md (hunk 1 above)
Comment (grant-arm) [reply apps/microtvm/ethosu/README.md]: Thanks @Mousius , I've fixed that now :)

File: apps/microtvm/ethosu/Makefile (hunk 2 above)
Comment (grant-arm) [reply apps/microtvm/ethosu/Makefile]: Good catch @Mousius . I've fixed that now.

**Review by Mousius — APPROVED** [review]
This looks good to me but someone else should merge given I contributed a fix to this branch :smile_cat:

**Review by manupa-arm — APPROVED** [review]
LGTM!

**Comment by manupa-arm** [comment]
Thanks @Mousius @grant-arm !
